### ISSUE TITLE
Segfault in bytecode but not in native code

### ISSUE THREAD

**Issue by zoggy** [ocaml/ocaml#10445]
Hello,

With ocaml 4.12.0, the following code ends with a segmentation fault when it is interpreted with the toplevel or compiled to bytecode, but not when it is compiled to native code:
````ocaml
class type t =
  object
    method label : string
  end

let p (o:t) = prerr_endline (Printf.sprintf "label=%S" o#label)

let f (o:t) =
  let (o:t) = object
      method label = o#label
    end
  in
  p o

class c : string -> t = fun label ->
  object
    method label = label
  end
let o = new c "my label"
let () = f o
````
````sh
$ ocaml ocamlbug.ml 
Erreur de segmentation (core dumped)
$ ocamlc -o t ocamlbug.ml  && ./t
Erreur de segmentation (core dumped)
$ ocamlopt -o t ocamlbug.ml  && ./t
label="my label"
````
Everything was ok with ocaml 4.06.1.

**Comment by gasche** [comment]
Thanks for the report!

Reproduction information: the bug occurred between 4.07.1 (works) and 4.08.0 (segfaults).

**Comment by Octachron** [comment]
This is a duplicate of #10324 which is fixed in trunk (and the 4.13 branch), see #10325 .

**Comment by zoggy** [comment]
Ok, thanks for your answers (and sorry for the noise).
